apps/ejabberd/src/http_upload/mod_http_upload.erl

```diff
@@ -202,29 +207,55 @@ generate_token(Host) ->
     base16:encode(crypto:strong_rand_bytes(token_bytes(Host))).
 
 
--spec file_too_large_error(MaxFileSize :: non_neg_integer()) -> jlib:exml().
-file_too_large_error(MaxFileSize) ->
+-spec file_too_large_error(MaxFileSize :: non_neg_integer(), Namespace :: binary()) -> jlib:exml().
+file_too_large_error(MaxFileSize, Namespace) ->
     MaxFileSizeBin = integer_to_binary(MaxFileSize),
     MaxSizeEl = #xmlel{name = <<"max-file-size">>, children = [exml:escape_cdata(MaxFileSizeBin)]},
     FileTooLargeEl = #xmlel{name = <<"file-too-large">>,
-                            attrs = [{<<"xmlns">>, ?NS_HTTP_UPLOAD}],
+                            attrs = [{<<"xmlns">>, Namespace}],
                             children = [MaxSizeEl]},
     Error0 = ?ERR_NOT_ACCEPTABLE,
     Error0#xmlel{children = [FileTooLargeEl | Error0#xmlel.children]}.
 
 
 -spec parse_request(Request :: exml:element()) ->
                            {Filename :: binary(), Size :: integer(),
-                            ContentType :: binary() | undefined} |
+                            ContentType :: binary() | undefined, Namespace :: binary()} |
                            bad_request.
 parse_request(Request) ->
-    Filename = exml_query:path(Request, [{element, <<"filename">>}, cdata]),
-    SizeBin = exml_query:path(Request, [{element, <<"size">>}, cdata]),
+    Namespace = exml_query:attr(Request, <<"xmlns">>),
+    Keys = [<<"filename">>, <<"size">>, <<"content-type">>],
+    [Filename, SizeBin, ContentType] =
+        case Namespace of
+            ?NS_HTTP_UPLOAD_025 -> [exml_query:path(Request, [{element, K}, cdata]) || K <- Keys];
+            ?NS_HTTP_UPLOAD_030 -> [exml_query:attr(Request, K) || K <- Keys]
+        end,
     Size = (catch erlang:binary_to_integer(SizeBin)),
-    ContentType = exml_query:path(Request, [{element, <<"content-type">>}, cdata]),
-
-    case is_binary(Filename) andalso <<>> =/= Filename
-        andalso is_integer(Size) andalso Size >= 0 of
+    case is_binary(Filename) andalso <<>> =/= Filename andalso is_integer(Size) andalso Size >= 0 of
```

Comment:
This conditional consists of 4 parts. I think it would more readable and easier to understand if
```erlang
is_binary(Filename) andalso <<>> =/= Filename andalso is_integer(Size) andalso Size >= 0
```
where encapsulated in a function with some meaningful name, f.e `is_filename_and_size_acceptable`.